### PR TITLE
opc-menu-drawer component

### PR DESCRIPTION
**Pull request by akhilmhdh**
# Closes

Closes 1331

# Explain the feature/fix

- Implemented opc-menu-drawer, one platform drawer component
- Added test for it.

## Does this PR introduce a breaking change

No

## Screenshots

<details>
<summary>View Screenshots</summary>
Screenshot 2021-07-26 at 11 56 51 AM
</details>

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] My changes generate no new warnings
- [x] I have performed a self-review of my own code
- [x] I have documented my code, particularly in areas like todo, complex logic, quick fix, temporary patch, etc.
#### If it is a new component
- [x] Does your component follow One Platform style guidelines?
- [x] Does your component web accessibility standards? [Helper Doc](https://www.w3.org/TR/wai-aria-1.1/)
- [x] Does your component support desktop screen sizes (350px, 720px, 1150px ,1920px)
#### Browsers you have tested in
- [ ] Latest two versions of Mozilla Firefox and Google Chrome supported by Red Hat Enterprise Linux distribution
- [x] Google Chrome [Latest 2 versions]
- [x] Mozilla Firefox [Latest 2 versions]
- [ ] Microsoft Edge [Latest 2 versions]
- [x] Apple Safari [Latest 2 versions]
- [ ] Microsoft Internet Explorer 11

### CODE REVIEW

**Review by ghanlohar — APPROVED** [review]
LGTM

**Comment by akhilmhdh** [comment]
Added better documentation...

**Comment by hybridx** [comment]
@akhilmhdh can you please rebase this code

**Comment by akhilmhdh** [comment]
@hybridx  Done

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+                    ]
+                },
+                {
+                    title: "BUILD in SERVICES",
+                    links: [
+                        { name: "blog", href: "#" },
+                        { name: "Documentation", href: "#" }
+                    ]
+                },
+        ];
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js:35]: Indendation needs to be fixed

File: packages/opc-menu-drawer/package.json [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "@one-platform/opc-menu-drawer",
+  "version": "0.0.1",
+  "description": "A web component based on Lit Element for Red hat One Platform Side Drawer.",
```
Comment (riginoommen) [inline packages/opc-menu-drawer/package.json:4]: Please update the description

File: packages/opc-menu-drawer/demo/index.html [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>opc-menu-drawer</title>
+    <script src="opc-menu-drawer.js"></script>
+</head>
+
+<body>
+    <button>Click to toggle drawer</button>
+    <opc-menu-drawer headerTitle="Akhil Mohan">
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/index.html:13]: Lets put a meaning full title which sets to context

File: packages/opc-menu-drawer/demo/index.html [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        <span slot="avatar">
+            AM
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/index.html:15]: Same as above

File: packages/opc-menu-drawer/test/opc-menu-drawer.test.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    expect(opcMenuDrawer.isOpen).toBeFalsy();
+    opcMenuDrawer.isOpen = true;
+    await opcMenuDrawer.updateComplete;
+    expect(opcMenuDrawer.isOpen).toBeTruthy();
+  });
+
+  it('should add headerTitle', async () => {
+    opcMenuDrawer.headerTitle = 'Akhil Mohan';
+    await opcMenuDrawer.updateComplete;
+    expect(opcMenuDrawer.headerTitle).toEqual('Akhil Mohan');
```
Comment (riginoommen) [inline packages/opc-menu-drawer/test/opc-menu-drawer.test.ts:61]: Update the test cases as per the conditions mentioned above

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      description: 'Header title of the drawer',
+    },
+  },
+};
+
+export const opcMenuDrawer = () => `
+<opc-menu-drawer isOpen headerTitle="Akhil Mohan"></opc-menu-drawer>
+<script>
+var links = [{
+                    title: "BUILD in SERVICES",
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js:22]:
```suggestion
                    title: "BUILT-IN SERVICES",
```

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+                    links: [
+                        { name: "blog", href: "#" },
+                        { name: "Documentation", href: "#" }
+                    ]
+                },
+                {
+                    title: "BUILD in SERVICES",
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js:29]:
```suggestion
                    title: "BUILT-IN SERVICES",
```

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+  },
+};
+
+export const opcMenuDrawer = () => `
+<opc-menu-drawer isOpen headerTitle="Akhil Mohan"></opc-menu-drawer>
+<script>
+var links = [{
+                    title: "BUILD in SERVICES",
+                    links: [
+                        { name: "blog", href: "#" },
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js:24]:
```suggestion
                        { name: "Blog", href: "#" },
```

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+                        { name: "Documentation", href: "#" }
+                    ]
+                },
+                {
+                    title: "BUILD in SERVICES",
+                    links: [
+                        { name: "blog", href: "#" },
```
Comment (riginoommen) [inline packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js:31]:
```suggestion
                        { name: "Blog", href: "#" },
```

**Review by riginoommen — CHANGES_REQUESTED** [review]

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js (hunk 1 above)
Comment (akhilmhdh) [reply packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js]: Done

File: packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js (hunk 6 above)
Comment (akhilmhdh) [reply packages/opc-menu-drawer/demo/opc-menu-drawer.stories.js]: Done

File: packages/opc-menu-drawer/package.json (hunk 2 above)
Comment (akhilmhdh) [reply packages/opc-menu-drawer/package.json]: Done

File: packages/opc-menu-drawer/package.json [hunk 10]
```diff
@@ -0,0 +1,48 @@
+{
+  "name": "@one-platform/opc-menu-drawer",
+  "version": "0.0.1",
+  "description": "opc-menu-drawer is a customizable drawer component webcomponent developed using Lit elements for Red Hat One Platform.",
```
Comment (riginoommen) [inline packages/opc-menu-drawer/package.json:4]:
```suggestion
  "description": "opc-menu-drawer is a customizable drawer web component which shows the menus and user profile",
```

**Review by riginoommen — APPROVED** [review]

**Review by deshmukhmayur — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Review by hybridx — APPROVED** [review]

**Pull request merged**